client/assets/js/board.js

```diff
@@ -84,4 +86,33 @@ export default class Board {
 		});
 	}
 
+    checkCollisionWithYourself() {
```

Comment:
The board does not have collision with itself. The method should rather be named something like "checkSnakeSelfCollisions" or something